### PR TITLE
(OraklNode) Docker image

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

## Docker files added

1. orakl-boot-api
2. orakl-node

## Minor updates

1. Update readme: updates .env description to reflect recent `eth_helper` and `klaytn_helper` updates
2. Add entrypoint for OraklNode: `./cmd/node/main.go`
3. Update reporter: set up `klaytn_helper` on start rather than setup.
-> through this change, reporter wallets will be refreshed on calling `/api/v1/reporter/refresh` from admin.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

## Summary by CodeRabbit

- **New Features**
	- Simplified database URL modification in Dockerfiles by using script execution commands.
	- Introduced Docker support for building Node.js applications with Golang for database migrations.
	- Added a new task `local:node` for running nodes locally.
	- Enhanced node setup with improved communication, administration, and reporting capabilities.
	- Enhanced Klaytn integration in the `reporter.go` file.

- **Bug Fixes**
	- Improved logging for invalid peer connections in the network.

- **Documentation**
	- Updated README with new task additions and provider URL changes.

### CODE REVIEW

**Review by jay-kim-kor — APPROVED** [review]
LGTM

File: dockerfiles/orakl-node.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+COPY --from=builder /app/node/migrations /app/migrations
+
+COPY --from=builder /app/node/nodebin /usr/bin
+
+CMD sh -c '_DATABASE_URL=$DATABASE_URL; if echo $_DATABASE_URL | grep -q "\?"; then \
+    _DATABASE_URL="${_DATABASE_URL}&sslmode=disable"; \
+    else \
+    _DATABASE_URL="${_DATABASE_URL}?sslmode=disable"; \
+    fi && \
+    migrate -database "$_DATABASE_URL" -verbose -path ./migrations up && nodebin'
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply dockerfiles/orakl-node.Dockerfile]: fixed from commit [6308a2e](https://github.com/Bisonai/orakl/pull/1305/commits/6308a2e1bdbbf971d4b9d55259e4a9328161ad50)

File: dockerfiles/start-go.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+_DATABASE_URL=$DATABASE_URL
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply dockerfiles/start-go.sh]: fixed from commit [2b0a21f](https://github.com/Bisonai/orakl/pull/1305/commits/2b0a21f18a80646a6f0c6b94917900a57287ee12)

File: dockerfiles/start-go.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+if echo $_DATABASE_URL | grep -q "\?"; then
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply dockerfiles/start-go.sh]: fixed from commit [2b0a21f](https://github.com/Bisonai/orakl/pull/1305/commits/2b0a21f18a80646a6f0c6b94917900a57287ee12)

File: dockerfiles/start-go.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    _DATABASE_URL="${_DATABASE_URL}&sslmode=disable"
+else
+    _DATABASE_URL="${_DATABASE_URL}?sslmode=disable"
+fi
+
+migrate -database "$_DATABASE_URL" -verbose -path ./migrations up
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply dockerfiles/start-go.sh]: fixed from commit [2b0a21f](https://github.com/Bisonai/orakl/pull/1305/commits/2b0a21f18a80646a6f0c6b94917900a57287ee12)

**Comment by nick-bisonai** [comment]
@coderabbitai please approve

**Comment by nick-bisonai** [comment]
> Review Status

all the comments have been resolved, approve this pr

**Pull request merged**